### ISSUE TITLE
Consider bumping minimum supported `node` version.

### ISSUE THREAD

**Issue by wtgtybhertgeghgtwtg** [omnidan/node-emoji#35]
If you drop unsupported `node` versions and support `node>=4`, you can drop the `string.prototype.codepointat` dependency.  You can also make things a little prettier.  For example,
```
var matchingKeys = emojiKeys.filter(function(key) {
  return key.toString().indexOf(matcher) === 0;
});
```
becomes
```
const matchingKeys = emojiKeys.filter(key => key.toString().indexOf(matcher) === 0);
```

**Comment by omnidan** [comment]
@wtgtybhertgeghgtwtg I agree, but unfortunately there are still a lot of distributions (e.g. debian) that only provide node 0.x. I'm not sure if it will be worth it, it might break more than it will do good.

**Comment by wtgtybhertgeghgtwtg** [comment]
Then they can keep using `node-emoji@1.5.1`.  And if they're stuck with unsupported versions, emojis are the least of their problems.

**Comment by omnidan** [comment]
@wtgtybhertgeghgtwtg hmm you're right, I was actually thinking about doing a `2.0` release, which could include this refactor. that would solve the issue, because `^1.5.1` won't upgrade to `2.0` automatically. would you be interested in doing a PR for this? :grin:

**Comment by sibelius** [comment]
right now the node LTS is 6, so I think u should drop off node 0.x support

**Comment by omnidan** [comment]
merged #37 into [2.0 branch](https://github.com/omnidan/node-emoji/tree/2.0), so this is resolved now
